src/api/verification.py

```diff
@@ -0,0 +1,157 @@
+            "all-expand",
+            "day-compact",
+            "day-expand",
+        ]:
+            logging.error(
+                f"The plot_option argument should either be all-compact, all-expand, day-compact, or day-expand, not {plot_option}."
+            )
+
+        if isinstance(fig_size, tuple):
+            if not (
+                (isinstance(fig_size[0], int) or isinstance(fig_size[0], float))
+                and (isinstance(fig_size[1], int) or isinstance(fig_size[1], float))
+            ):
+                logging.error(
+                    "The fig_size argument should be a tuple of integers or floats."
+                )
+                return None
+        elif not fig_size is None:
+            logging.error(
+                f"The fig_size argument should be a tuple of integers or floats. Here is the variable type that was passed {type(fig_size)}."
+            )
+            return None
+
+        if (isinstance(num_threads, int) and num_threads < 1) or (
+            not isinstance(num_threads, int)
+        ):
+            logging.error("The number of threads should be an integer greater than 1.")
+
+        self.output_path = output_path
+        self.lib_items_path = lib_items_path
+        self.plot_option = plot_option
+        self.fig_size = fig_size
+        self.num_threads = num_threads
+
+    def run_single_verification(self, case: dict = None) -> None:
+        """Run a single verification and generate a markdown report of the results
+
+        Args:
+            case (dict): Verification case dictionary.
+        """
+        # Input validation
+        if case is None:
+            logging.error("A case must be passed as an argument.")
+
+        if not isinstance(case, dict):
+            logging.error(
+                f"A case dictionary must be passed as an argument, not a {type(case)}."
+            )
+
+        # Run verification
+        items = assemble_verification_items(
+            cases=case, lib_items_path=self.lib_items_path
+        )
+        results = run_libcase(
+            item_dict=items[0],
+            plot_option=self.plot_option,
+            output_path=self.output_path,
+            fig_size=self.fig_size,
+            produce_outputs=True,
+        )
+
+        # Output case summary
+        cases_file = f"{self.output_path}/{case['no']}.md"
+        with open(cases_file, "w") as fw:
+            fw.write(results[list(results.keys())[0]]["md_content"])
+
+    def run(self) -> None:
+        """Run verification and generate a markdown report of the results."""
+        # Input validation
+        if self.output_path is None:
+            self.output_path = ""
+        if self.cases is None or len(self.cases) == 0:
+            logging.error(
+                "The verification case suite is empty, there is nothing to run."
+            )
+            return None
+
+        # Run verifications
+        with multiprocessing.Pool(len(self.cases)) as c:
```

Comment:
It seems like ```num_threads``` variable isn't used anywhere and I'm assuming it could be used here? Could you check this?

Reply:
Thanks for catching that! It used to be include but somehow did not make it's way here.